lib/routes/routeBackbeat.js

```diff
@@ -650,6 +667,45 @@ function putMetadata(request, response, bucketInfo, objMd, log, callback) {
                     return next();
                 });
             },
+            async () => {
+                // If we create a new version of an object (so objMd is null),
+                // we should make sure that the masterVersion is versionned.
+                // If an object already exists, we just want to update the metadata
+                // of the existing object and not create a new one
+                if (versioning && !objMd) {
+                    let masterMD;
+
+                    try {
+                        masterMD = await metadata.getObjectMDPromised(bucketName, objectKey, {}, log);
+                    } catch (err) {
+                        if (err.is?.NoSuchKey) {
+                            log.debug('no master found for versioned object', {
+                                method: 'putMetadata',
+                                bucketName,
+                                objectKey,
+                            });
+                        } else {
+                            throw err;
```

Comment:
I'm not sure we want to throw, the current stack is not using promises yet, so we tend to use more errors and calling the callbacks with it. I fear that a throw here will make the process crash, as we do not try/catch our code?


```suggestion
                            return next(err);
```

Reply:
We can throw safely. We are in an `async` process. Async is able to switch from 'async/await' to 'callback'. Async will transform this throw in a callback one 🙏 

```
import async from "async";

function main() {
  async.series(
    [
      function (callback) {
        setTimeout(function () {
          console.log("Task 1");
          callback(null, "one");
        }, 200);
      },
      async function () {
        await new Promise((resolve) => {
          setTimeout(function () {
            console.log("Task 1.5");
            resolve();
          }, 150);
        });

        throw new Error("Error in Task 1.5");
      },
    ],
    function (err, results) {
      console.error(err);
      console.log(results);
    },
  );
}

main();
```

But I'm open minded as you mention that the current stack don't use promises yet. I would like to start to change that and move to a more modern stack by starting async/await syntax, in parallel of the SDK migration that introduce a lot async/await. Let me know